### ISSUE TITLE
Dirty "regions" instead of dirty lines?

### ISSUE THREAD

**Issue by davidbrochart** [selectel/pyte#172]
The [dirty](https://github.com/selectel/pyte/blob/259ee027b4d6061c9a241c6971d1eaf0fd2376da/pyte/screens.py#L151) screen attribute is a set of line numbers which should be re-drawn. I'm using `pyte` in a Textual application where one can decide to [update a region](https://textual.textualize.io/guide/widgets/#region-updates), where a [Region](https://textual.textualize.io/api/geometry/#textual.geometry.Region) is a rectangle.
I can of course say that for each dirty line corresponds a dirty region that has the width of the screen and a height of 1, but it would be more optimal if I knew what (segments of) characters are dirty inside each dirty line. I'm wondering if `pyte` could have something similar, i.e. dirty regions instead of just dirty lines?

**Comment by superbobry** [comment]
Yeah, you can fork `DiffScreen` and implement your own version tracking dirty regions. IIRC we went with lines because it was simpler and sufficient for our use-cases.
